src/crossroads/main.py

```diff
@@ -19,86 +20,115 @@
-
-class MainForm(Form):
-    first_name_field = TextField('Your name', description='Put here your name',
-                        validators=[Required()])
-    first_raw_coordinates = FileField('Upload')
-    second_name_field = TextField('Your friend\'s name', description='Put here name of your friend',
-                       validators=[Required()])
-    second_raw_coordinates = FileField('Upload')
+class HomeForm(FlaskForm):
+    first_name_field = StringField('Your name', description='Put here your name',
+                                   validators=[DataRequired(), Length(min=2, max=30)])
+    first_raw_coordinates = FileField('Upload',
+                                      validators=[FileRequired(), FileAllowed(['json'], 'Just Google\'s json!')])
+    second_name_field = StringField('Your friend\'s name', description='Put here name of your friend',
+                                    validators=[DataRequired(), Length(min=2, max=30)])
+    second_raw_coordinates = FileField('Upload',
+                                       validators=[FileRequired(), FileAllowed(['json'], 'Just Google\'s json!')])
     submit_button = SubmitField('Find crossed roads!')
 
-    def validate_hidden_field(form, field):
-        raise ValidationError('Always wrong')
-
 @app.route('/', methods=['GET', 'POST'])
 def main_page():
-    # Just for tests in order to create coordinates on map
-    coordinates = generate_coordinates()
-    form = MainForm()
-    map = generate_map(coordinates)
-    return render_template('index.html', form = form, crmap = map)
+    form = HomeForm()
+    coordinates = sample_coordinates()
+    google_map = generate_map(coordinates)
+
+    if form.validate_on_submit():
+        if request.method == 'POST':
+            user1 = request.files['first_raw_coordinates']
+            first_user_raw_data = user1.read()
+            first_user_name = request.form['first_name_field']
+
+            user2 = request.files['second_raw_coordinates']
+            second_user_raw_data = user2.read()
+            second_user_name = request.form['second_name_field']
+
+            if validate_json(first_user_raw_data) and validate_json(second_user_raw_data):
+                return locations_page(first_user_raw_data, second_user_raw_data, first_user_name, second_user_name)
+            else:
+                flash('Just Google\'s json!')
+                return render_template('index.html', form=form, crmap=google_map,)
+
+    return render_template('index.html', form=form, crmap=google_map, )
+
+def locations_page(first_user_raw_loc, second_user_raw_loc, first_user_name, second_user_name):
+    coordinates = generate_coordinates(first_user_raw_loc, second_user_raw_loc)
+    google_map = generate_map(coordinates)
+
+    return render_template('locations.html', crmap=google_map, name1=first_user_name, name2=second_user_name)
 
 def main():
-    app.run(host='0.0.0.0', port=8080)
+    app.run(host='localhost', port=8080)
```

Comment:
Leave host with `0.0.0.0` since with this IP address it will work directly on host and docker